### PR TITLE
use SpecialFolder.UserProfile in few places

### PR DESCRIPTION
**Pull request by kasperk81**
same as CliFolderPathCalculatorCore code:
 https://github.com/dotnet/sdk/blob/c3a8f72c3a5491c693ff8e49e7406136a12c3040/src/Common/CliFolderPathCalculatorCore.cs#L34-L38

### CODE REVIEW

File: src/Cli/Microsoft.TemplateEngine.Cli/Commands/CliPathInfo.cs [hunk 1]
```diff
@@ -42,6 +42,7 @@ private static string GetUserProfileDir(IEnvironment environment) => environment
             RuntimeInformation.IsOSPlatform(OSPlatform.Windows)
                 ? "USERPROFILE"
                 : "HOME")
+            ?? Environment.GetFolderPath(Environment.SpecialFolder.UserProfile)
```
Comment (marcpopMSFT) [inline src/Cli/Microsoft.TemplateEngine.Cli/Commands/CliPathInfo.cs:45]: Would it make sense to just call GetDotnetHomePath here? I believe template engine cli already has a reference to Microsoft.DotNet.Configurer so it should be available.

**Comment by marcpopMSFT** [comment]
Makes sense. I guess we'll take the duplication for now.

@dotnet/razor-compiler can you double check the change to the RazorSDK?

File: src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs [hunk 2]
```diff
@@ -77,13 +77,7 @@ public static string GetTempPath(string workingDir)
...
             }
 
-            var userProfile = Environment.GetEnvironmentVariable("USERPROFILE");
-            if (Path.IsPathRooted(userProfile))
-            {
-                return userProfile;
-            }
-
-            return Environment.GetEnvironmentVariable("SYSTEMROOT");
+            return Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
```
Comment (jjonescz) [inline src/RazorSdk/Tool/ServerProtocol/ServerConnection.cs:80]: I don't understand the motivation for this change. The PR description references `GetDotnetHomePath` but here we are resolving a temp path, so that seems unrelated. I see `SpecialFolder.UserProfile` is equivalent to `%USERPROFILE%`, but it's not clear to me why we removed the `%SYSTEMROOT%` fallback?

**Review by jjonescz — APPROVED** [review]

**Review by marcpopMSFT — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Comment by marcpopMSFT** [comment]
Thanks for the contribution.
